frontend/src/metabase/dashboard/hoc/withAutoApplyFilters.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useCallback, useEffect, useState } from "react";
+import _ from "underscore";
+
+import { ApplyButton } from "./withAutoApplyFilters.styled";
+
+export default function withAutoApplyFilters<T>(
+  Component: React.ComponentType<T>,
+) {
+  interface WithAutoApplyFiltersProps {
+    // XXX: Fix types
+    dashboard: any;
+    parameterValues: any;
+    setParameterValue: any;
+    setParameterValues: any;
+    fetchDashboardCardData: any;
+  }
+
+  return function WithAutoApplyFilters(props: T & WithAutoApplyFiltersProps) {
+    const {
+      dashboard,
+      parameterValues,
+      setParameterValue,
+      setParameterValues,
+      fetchDashboardCardData,
+    } = props;
+
+    // XXX: control parameterValues and dashboard to be changed at the same time, otherwise
+    // the logic inside <Dashboard />'s `componentDidUpdate` will be triggered twice,
+    // causing the dashboard cards to be fetched twice, when there'are parameters
+    // in the query params.
+    const [state, setState] = useState<{
```

Comment:
I need to control both `parameterValues`, and `dashboard` passed to `<Dashboard />` to be changed at the same time, otherwise the logic inside <Dashboard />'s `componentDidUpdate` will be triggered twice, causing the dashboard cards to be fetched twice, when there are parameters in the query params.